assets/src/scripts/components/graph/state/cursor.spec.js

```diff
@@ -25,23 +25,29 @@ describe('graph component cursor state', () => {
         expect(getCursor(store.getState())).toBe(cursorDate);
     });
 
-    it('getCursor returns right extent on null cursor', () => {
+    it('getCursor returns viewport end date on null cursor', () => {
+        const viewport = {endDate: new Date('2010-10-10')};
+        const cursorDate = getCursor.resultFunc(null, viewport);
```

Comment:
Ok, we disagree on this one. From my standpoint, this unit test is checking only that the `getCursor` selector returns the end date of the viewport or domain, and doesn't need to be concerned with the implementation details of `getViewport` and `getDomain`... we've got separate tests for those.

I feel the mocks got out of hand with WDFN because we mocked up so much 2nd, 3rd, or 4th-degree state. In cases like this, do we gain anything by mocking dependent selector state up vs just using `resultFunc`?

Reply:
We may need to take this to the group. In my view, the test should test all of the code. In this cause we are using a viewport selector within the selector to get the information, but that might change too. I don't like the pattern of using implementation detail in the unit test and in most selector test cases it's not necessary. Maybe we can have a side discussion after stand-up.